### PR TITLE
Pr 9180

### PR DESCRIPTION
**Pull request by todb-r7**
Updated after @bcoles's review over on rapid7/metasploit-framework#9180

One more test, @RootUp ! Mostly to check that the conditionals actually fall through correctly.

After that, I'll muddle through the markdown documentation, unless you want to give that a shot, as described here:

https://github.com/rapid7/metasploit-framework/wiki/Generating-Module-Documentation

Thanks for your patience on this.

### CODE REVIEW

File: modules/auxiliary/gather/samsung_browser_sop_bypass.rb [hunk 1]
```diff
@@ -118,11 +119,10 @@ def collect_data(request)
     cred = JSON.parse(request.body)
     u = cred['user']
     p = cred['pass']
-    if not u.nil? and not u.empty? and not p.nil? and not p.empty?
-      print_good("#{cli.peerhost}: Collected credential for '#{datastore['TARGET_URL']}' #{u}:#{p}")
-      store_cred(u,p)
```
Comment (bcoles) [inline modules/auxiliary/gather/samsung_browser_sop_bypass.rb]: Was `store_cred(u,p)` supposed to have been removed ?

**Comment by RootUp** [comment]
Hi @todb-r7 and @bcoles 
While reload_all it generates the below error.

`
msf > reload_all 
[*] Reloading modules from all module paths...
[-] WARNING! The following modules could not be loaded!
[-] 	/opt/metasploit/apps/pro/vendor/bundle/ruby/2.3.0/gems/metasploit-framework-4.15.4/modules/auxiliary/gather/samsung_browser_sop_bypass.rb: SyntaxError /opt/metasploit/apps/pro/vendor/bundle/ruby/2.3.0/gems/metasploit-framework-4.15.4/modules/auxiliary/gather/samsung_browser_sop_bypass.rb:52: syntax error, unexpected tIDENTIFIER, expecting ']'
...) function. Use the variable 'x' to refer to the new tab.',
...                               ^
/opt/metasploit/apps/pro/vendor/bundle/ruby/2.3.0/gems/metasploit-framework-4.15.4/modules/auxiliary/gather/samsung_browser_sop_bypass.rb:54: syntax error, unexpected ']', expecting keyword_end
    ])
     ^
/opt/metasploit/apps/pro/vendor/bundle/ruby/2.3.0/gems/metasploit-framework-4.15.4/modules/auxiliary/gather/samsung_browser_sop_bypass.rb:143: syntax error, unexpected end-of-input, expecting keyword_end
`

**Comment by todb-r7** [comment]
Oops, yeah I screwed it up. My fault for not testing before pushing!

Will fix in a bit.

**Comment by todb-r7** [comment]
Fixed, @RootUp !

**Pull request merged**